sdk/storage/Azure.Storage.Files.Shares/samples/SharesSampleTest.cs

```diff
@@ -0,0 +1,259 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using NUnit.Framework;
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.IO;
+using Azure.Storage.Files.Shares.Models;
+using Azure.Storage.Files.Samples;
+
+namespace Azure.Storage.Files.Shares.Samples.Tests
+{
+    /// <summary>
+    /// Turn the simple Shares samples into tests.
+    /// </summary>
+    public class SharesSampleTest : SampleTest
+    {
+        [Test]
+        public void Sample01a_HelloWorld_Upload()
+        {
+            // Create a temporary Lorem Ipsum file on disk that we can upload
+            string localFilePath = CreateTempFile(SampleFileContent);
+
+            string shareName = Randomize("sample-share");
+            ShareClient share = new ShareClient(ConnectionString, shareName);
+            try
+            {
+                Sample01a_HelloWorld.Upload(ConnectionString, shareName, localFilePath);
+            }
+            finally
+            {
+                share.Delete();
+                File.Delete(localFilePath);
+            }
+        }
+
+        [Test]
+        public async Task Sample01b_HelloWorldAsync_UploadAsync()
```

Comment:
For a lot of these tests we aren't asserting anything, whereas previously we were. I wonder if it isn't better to just leave the existing tests in place and strategically place the snippet region around the code we want in the snippet, thereby leaving in our assertions.

Reply:
We’re validating that the sample code runs without exceptions.  Where possible we still assert.  

People still read these files too and I don’t think it’s helpful to have a bunch of asserts in there when someone is trying to learn a new API.

Reply:
My main issue with this is that the assertions are in a separate file that in some cases depend on inspecting the static method being called, which seems prone to break. Also, in order to even do the assertions, we need to recreate the client using the same passed in parameter values, which also seems brittle. Not worth holding up the PR though.

Reply:
It's what do you want to optimize for - test maintainability or customer understanding.

Reply:
I think it would be better to just discard the assertions completely then. Also, I think the fact that the methods now take parameters slightly decreases customer understanding.